objects/obj_creation/Create_0.gml

```diff
@@ -73,8 +73,9 @@ fleet_type=1;
 strength=5;cooperation=5;
 purity=5;stability=5;
 var i;i=-1;repeat(6){i+=1;adv[i]="";adv_num[i]=0;dis[i]="";dis_num[i]=0;}
-homeworld="Temperate";homeworld_name=scr_star_name();
-recruiting="Death";recruiting_name=scr_star_name();
+var starNameGenerator = new StarNameGenerator();
+homeworld="Temperate";homeworld_name=starNameGenerator.generate_random_name();
+recruiting="Death";recruiting_name=starNameGenerator.generate_random_name();
```

Comment:
hmm, the generator would be easier to do, if I didn't have to instantiate it, just have a 
`StarNameGenerator.generate_random_name()` calls. Not sure how to implement it in GML.

Reply:
This would just be accomplished via making `generate_random_name` a static, issue is that you still have to init the static by calling the function or constructor at least once prior to being able to have that syntax style. I would just make `StarNameGenerator` a non constructor function and define that random generator as a static of the function so that `StarNameGenerator` basically acts as a namespace, and then just invoke it to initialize the statics.

Reply:
Scratch the above, since the goal is to make a generic way to implement the functionality for other areas where names are provided.

Reply:
ok, I settled with a global `name_generator` struct, which exposes various `generate_imperial_name()` etc. functions.